src/WindowTitle.ps1

```diff
@@ -58,3 +58,45 @@ function Set-WindowTitle {
         }
     }
 }
+
+function Set-TabTitle {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseShouldProcessForStateChangingFunctions", "")]
+    param($GitStatus)
+    $settings = $global:GitPromptSettings
+
+    if ($settings.TabTitle -eq $false) {
+        return
+    }
+
+    # If the user is running Powershell ISE then name the tab
+    if($psISE -and $GitStatus){
+        $existingTabNames = $psISE.PowerShellTabs | % {$_.DisplayName}
+        $currentTabName = $psise.CurrentPowerShellTab.DisplayName
+        $tabName = Get-TabTitle $GitStatus $existingTabNames $currentTabName
+        $psise.CurrentPowerShellTab.DisplayName = $tabName
+    }
+}
+
+function Get-TabTitle {
```

Comment:
I think we change this to `Get-UniqueTableTitle` with a single parameter of `$Title` and this function simply ensures that $Title is unique and if not, it will append numbers until a unique title is found.

Reply:
👍 

I think you can simplify a bit, too:

```powershell
function Get-UniqueTabDisplayName ($Name, $Format="{0} {1}") {
  $tabNumber = 1
  $names = $psISE.PowerShellTabs | Select-Object -ExpandProperty DisplayName
  $newName = $Name
  while ($names -Contains $newName) {
    # TODO: should this format also be in $GitPromptSettings?
    $newName = $Format -f $Name,$tabNumber++
  }
  return $newName
}
```

Note I dropped the parentheses to align with ISE's default numbering scheme.

Reply:
I think I tried it like this initially but noticed an edge case where if you open a few tabs to the same repo i.e. posh-git [master], posh-git [master] (1), posh-git [master] (2) but then you close tab  posh-git [master] (1) and open a new tab at the same repo it tries to create a new tab called posh-git [master] (2) and fails.  Hence the logic to get the highest number that has been assigned so far.

Reply:
I believe this correctly fills in the gaps:
![image](https://user-images.githubusercontent.com/133987/44303370-deebeb80-a304-11e8-9fb5-048b36123a52.png)
![image](https://user-images.githubusercontent.com/133987/44303349-553c1e00-a304-11e8-81a2-282132dcbda9.png)
![image](https://user-images.githubusercontent.com/133987/44303364-b9f77880-a304-11e8-9080-7c0f6bfa0d2c.png)

That said, this does open the possibility of changing branches/tabs such that you get:

```
[ "posh-git [master]", "posh-git [master] (1)", "posh-git [master] (2)"]
[ "posh-git [master]", "posh-git [george]", "posh-git [master] (2)"]
[ "posh-git [master]", "posh-git [george]", "posh-git [george] (1)"]
[ "posh-git [george] (2)", "posh-git [george]", "posh-git [george] (1)"]
```

As opposed to:


```
[ "posh-git [master]", "posh-git [master] (1)", "posh-git [master] (2)"]
[ "posh-git [master]", "posh-git [george] (1)", "posh-git [master] (2)"]
[ "posh-git [master]", "posh-git [george] (1)", "posh-git [george] (2)"]
[ "posh-git [george]", "posh-git [george] (1)", "posh-git [george] (2)"]
```

Reply:
You're right this is much better and simpler.  I hadn't looked at your looping logic properly I thought it was simply counting the number of tabs with the same name and using that count to name the tab. Wrist slapped.
Personally, I don't really think it matters what the number is as long as it's unique and doesn't change for a given tab.